### DISCUSSION TITLE
Multiple Vcenters to same NetBox site

### DISCUSSION THREAD

**Discussion by srilaxmiwas** [bb-Ricardo/netbox-sync#324] [General]
Hello all,

Can you please help me, how to add multiple vcenter's to the same NetBox site. 

Thanks

**Comment by bb-Ricardo** [comment]
Hi,

you have to use this setting: 
https://github.com/bb-Ricardo/netbox-sync/blob/b9e236df43c463d4d09a8afeeb6d69f3f8ee62aa/settings-example.ini#L177-L187

You have to define each cluster which belongs to the same site in each vCenter section. That should do the trick.

**Comment by srilaxmiwas** [comment]
I have one environment/site with 4 vcenters with different IP addresses, usernames and passwords. can you please provide me one example with one site, different vcenters, usernames and passwords.

Thanks

**Reply by bb-Ricardo** [reply]
You have to add each vcenter as a separate source. And if all resources belong to the same site you could configure `cluster_site_relation = .* = site name` for each source.

**Comment by srilaxmiwas** [comment]
Example:  In below example when I add only 1*vcenter it is working fine but when add 2nd vcenter, getting below error:
2023-07-14 01:21:54,549 - ERROR: Unable to connect to vCenter instance 'vcenter1, vcenter2' on port 443. Reason: [Errno -2] Name or service not known
2023-07-14 01:21:54,550 - INFO: Source 'vcenter1-Cluster1' is currently unavailable. Skipping
2023-07-14 01:21:54,550 - ERROR: No working sources found. Exit.

;;; [source/*]
;;;
;;; Controls the parameters of a defined source. The string past the slash will be used as
;;; a sources name. Sources can be defined multiple times to represent different sources.
;;;

[source/vcenter1-Cluster1],[source/vcenter1-Cluster2],[source/vcenter2-Cluster1]

; Defines if this source is enabled or not
enabled = True

; type of source. This defines which source handler to use
type = vmware

; host name / IP address of the vCenter
host_fqdn = vcenter1, vcenter2

; TCP port to connect to
port = 443

; username to use to log into vCenter
username = administrator, administrator

; password to use to log into vCenter
password = secret, secret

**Reply by bb-Ricardo** [reply]
Hi,

Almost 😄:

```
[source/vcenter1]

host_fqdn = vcenter1

[source/vcenter2]

host_fqdn = vcenter2

```

Just redefine a new source for each vcenter. Make sure the settings are configured in each source section.

**Comment by srilaxmiwas** [comment]
I am looking for a solution to add multiple Vcenter IP addresses and their username and passwords to single settings.ini file. 

Thanks
